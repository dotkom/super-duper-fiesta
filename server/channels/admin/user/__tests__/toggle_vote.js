jest.mock('../../../../models/user');
jest.mock('../../../../utils');
const { toggleCanVote } = require('../toggle_vote');
const { emit, broadcast } = require('../../../../utils');
const { updateUserById } = require('../../../../models/user');
const { generateUser, generateSocket } = require('../../../../utils/generateTestData');

describe('toggleCanVote', () => {
  beforeEach(() => {
    updateUserById.mockImplementation(
      async (userid, data) => generateUser({ id: userid, ...data }));
  });

  const generateData = () => ({
    canVote: false,
  });

  it('emits toggled can vote action on success', async () => {
    await toggleCanVote(generateSocket(), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });
});
