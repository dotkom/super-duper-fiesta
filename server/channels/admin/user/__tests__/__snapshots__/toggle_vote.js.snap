// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleCanVote emits toggled can vote action on success 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": false,
          "genfors": "1",
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "TOGGLE_CAN_VOTE",
    Object {
      "_id": "123",
      "canVote": false,
      "completedRegistration": false,
      "genfors": "1",
      "id": undefined,
      "name": "Namy",
      "onlinewebId": "123",
      "permissions": 5,
    },
  ],
]
`;
